src/util/path/index.ts

```diff
@@ -755,81 +866,76 @@ export const getPointOnPath = (
-
-  for (let i = 0, len = path.length; i < len; i++) {
-    const currentPath = path[i];
-    const coordsStr = currentPath.slice(1).trim();
-    const coords = [];
-    let command = currentPath.charAt(0);
-    const coordsParsed = [command];
-
-    if (command.toLowerCase() === 'a') {
-      // arcs have special flags that apparently don't require spaces so handle special
-      for (let args; (args = regArcArgumentSequence.exec(coordsStr)); ) {
-        for (let j = 1; j < args.length; j++) {
-          coords.push(args[j]);
-        }
-      }
-    } else {
-      let match;
-      while ((match = re.exec(coordsStr))) {
-        coords.push(match[0]);
-      }
-    }
+export const parsePath = (pathString: string): TComplexPathData => {
+  // clean the string
+  // add spaces around the numbers
+  pathString = pathString
+    .replace(new RegExp(`(${numberRegExStr})`, 'gi'), ' $1 ')
+    // replace annoying commas and arbitrary whitespace with single spaces
+    .replace(/,/gi, ' ')
+    .replace(/\s+/gi, ' ');
 
-    for (let j = 0, jlen = coords.length; j < jlen; j++) {
-      const parsed = parseFloat(coords[j]);
-      if (!isNaN(parsed)) {
-        coordsParsed.push(parsed);
+  const res: TComplexPathData = [];
+  for (const match of pathString.matchAll(rePathCommand)) {
+    let matchStr = match[0];
+    const chain: TComplexPathData = [];
+    let paramArr: RegExpExecArray | null;
+    do {
+      paramArr = new RegExp(rePathCommand.source, 'id').exec(matchStr);
+      if (!paramArr) {
+        break;
       }
-    }
-
-    const commandLength = commandLengths[command.toLowerCase()],
-      repeatedCommand = repeatedCommands[command] || command;
-
-    if (coordsParsed.length - 1 > commandLength) {
-      for (
-        let k = 1, klen = coordsParsed.length;
-        k < klen;
-        k += commandLength
-      ) {
-        result.push([command].concat(coordsParsed.slice(k, k + commandLength)));
-        command = repeatedCommand;
+      const filteredGroups = paramArr.filter((g) => g);
+      // indices is a JS regex thing that apparently isn't nicely supported by TS (set from the 'd' flag)
+      const filteredIndices = (paramArr as any).indices.filter((i: never) => i);
+      // remove the first element from the match array since it's just the whole command
+      filteredGroups.shift();
+      filteredIndices.shift();
+      // if we can't parse the number, just interpret it as a string
+      // (since it's probably the path command)
+      const command = filteredGroups.map((g) => {
+        const numParse = Number.parseFloat(g);
+        if (Number.isNaN(numParse)) {
+          return g;
+        } else {
+          return numParse;
+        }
+      });
+      chain.push(command as any);
+      // chop off that last set of params
+      if (filteredIndices.length <= 1) {
+        break;
       }
-    } else {
-      result.push(coordsParsed);
-    }
+      matchStr = matchStr.substring(0, filteredIndices[1][0]);
```

Comment:
what is this doing? to me is keeping the first part of the matchString, so basicaly the command like 'Q', 'M'...
But then what are we doing with it?
Why do we think we need this?

Has anything to do with the fact that those regex return always an array of 40 items to me, and i have no idea why?
Is it like that if the string is long we need more parses?

Reply:
It's for chained commands